### DISCUSSION TITLE
Non-official ICS files wanted in new Service Provider?

### DISCUSSION THREAD

**Discussion by johanwannheden** [mampfes/hacs_waste_collection_schedule#2104] [Q&A]
Hi!
I stumbled on this project and would like to contribute! For some time, I've generated an ICS file based on a PDF file my municipality provides as an official document. Hence my question is: if I wanted to contribute a new service provider for my municipality, would it be acceptable to do so with a reference to the ICS file I provide myself? (I could host it in a public Github repository or similar).
Thanks
Johan

**Comment by jamesmacwhite** [comment]
I have exactly the same question! My council publishes a PDF file or provides a service for a reminder via email on each bin collection day but no ICS/iCal calendar. Every year I always create a Google Calendar and make it public on the off chance anyone else searches for it, but this could serve the basis for use in this integration, and this would provide the data in a consistent format and it can easily be provided as iCal through Google Calendar. Although this could be hosted anywhere required, providing it is generated by a source.

The only downside is the reliance on doing the translation each year and any specific mapping needed in this plugin to take the data properly. Although being able to control the calendar format/schedule may be more beneficial for this side, just it will remain a manual conversion of a PDF would need to occur in my case every year.

**Comment by 5ila5** [comment]
There currently is no streamlined way to collect collection dates, that cannot be parsed from a website, API or ICS file.

But we have some sources marked as unofficial which parses dates from a third party. 

You could publish the ics somewhere and add a [ICS source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_ics.md), as long as it's clear that the collection dates do not come directly from the source but from a third party

**Reply by jamesmacwhite** [reply]
Thanks for this. That seems fine and totally understand things are tricky without an API/calendar format. In the case of publishing a third-party ICS source. I am looking to do this. I created a repository and an ical folder within (I will be adding more):

https://github.com/jamesmacwhite/gedling-borough-council-bin-calendars

I added a disclaimer at the bottom about the unofficial nature.

Would this be acceptable for inclusion once an ICS source was submitted as pull request?

**Reply by 5ila5** [reply]
LGTM
just so you know, if you want that people can just add the url and do not need to update the files you should either keep the links static (do not include year) or only include the year (yyyy) as this will get automatically replaced when replacing the year with `{%Y}` in the config

**Reply by jamesmacwhite** [reply]
Thanks, adding in the year is more for keeping track of things. I did think that adding the year might be an issue for updates. I'll review that, so it's more flexible. Thinking about it, dynamically using the year value is still going to be problematic, as the new calendars get released towards the end of the year e.g. for 2025, this would be released at around November 2024.

I'll look at adding these into the integration as sources, once I've refactored the naming to be less problematic for updating.
